### PR TITLE
Allow custom configuration

### LINKED ISSUE #135: Allow to load custom configuration for the description
### Context

Nowadays the configuration files that describe the robot are located at:
https://github.com/Ekumen-OS/andino/tree/humble/andino_description/config/andino

However if the user wanted to have it a bit different; `motors` added in a different manner, `lidar` displaced and connected somwhere else, `base` with different size; they should create their own URDF files as the configuration files are harcoded within the [`xacro` file](https://github.com/Ekumen-OS/andino/blob/humble/andino_description/urdf/andino.urdf.xacro)

### Proposal
- Modify the xacro to receive by argument the path to the configuration files
- Use as default the current files if no argument is passed down to the xacro files.
- Modify the launch file to accept passing the path to the configuration yaml.

### Also!
 - Update documentation

### PR DESCRIPTION
**Pull request by JesusSilvaUtrera**
# 🎉 New feature

Closes #135  and #136 

## Summary
I have continued the work on these issues to add the possibility to pass arguments to the xacro file directly from the launch file. I created a `hardware.yaml` in the `config` folder to load the params from there instead of hardcoding them in the `andino_control.xacro` file.

I have also updated the `README.md` with these new features.

## Test it
You can test that it works running `ros2 launch andino_description view_andino.launch.py`, which executes RViz and the `robot_state_publisher` to check that everything remains as before, but now more parameterized.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by JesusSilvaUtrera** [comment]
The arguments for the xacro are being passed directly inside the launch file, not as arguments of the launch files (I don't know if this was the expected result, IMO it is easier to understand this way, just changing the path in the launch file).

File: andino_description/urdf/include/andino_control.urdf.xacro [hunk 1]
```diff
@@ -1,34 +1,43 @@
...
-      <state_interface name="position"/>
-    </joint>
-  </ros2_control>
+<!-- ===================== Control xacro =========================================
+
+  params:
+  - hardware_props [dictionary]: hardware properties loaded from the YAML file.
+-->
+
+  <xacro:macro name="control" params="hardware_props" >
```
Comment (francocipollone) [inline andino_description/urdf/include/andino_control.urdf.xacro:10]: I think that you are missing to use the props in the file

**Review by francocipollone — COMMENTED** [review]
Thanks for the contribution

**Comment by francocipollone** [comment]
> The arguments for the xacro are being passed directly inside the launch file, not as arguments of the launch files (I don't know if this was the expected result, IMO it is easier to understand this way, just changing the path in the launch file).

That's ok and also expected behavior. 
From a user perspective it is useful to be able to indicate other .yaml file to be used instead the default one, however the parameters are always read from a yaml file.

File: andino_description/urdf/include/andino_control.urdf.xacro (hunk 1 above)
Comment (JesusSilvaUtrera) [reply andino_description/urdf/include/andino_control.urdf.xacro:10]: Oops, my bad, I thought I have changed it but it was only on my mind jaja

File: andino_description/README.md [hunk 2]
```diff
@@ -5,6 +5,14 @@ This package holds the urdf description of the robot.
 
 <img src="docs/robot_rviz.png">
 
+In the `urdf` folder you have the URDF files that contain the description of the robot, divided in different modules and merged into `andino.urdf.xacro` file.
+
+## Configuration
+
+In case you want to change the physical properties of some of the components of the robot, you can do it modifying the default YAML files inside the `config/andino` folder.
+
+You can even add your own configuration files in another directory in the `config` folder, and passing this directory to the main file using `yaml_config_dir` xacro argument inside the launch files.
```
Comment (jballoffet) [inline andino_description/README.md:14]: nit: passing -> pass

Comment (jballoffet) [inline andino_description/README.md:14]: nit: "using the `yaml_config_dir` xacro argument on the launch files."

File: andino_description/launch/andino_description.launch.py [hunk 3]
```diff
@@ -41,15 +41,16 @@
 
 def generate_launch_description():
 
+    # Obtains andino_description's share directory path.
+    pkg_andino_description = get_package_share_directory('andino_description')
```
Comment (jballoffet) [inline andino_description/launch/andino_description.launch.py:45]: Consider placing this closer to where it is required.

File: andino_description/launch/view_andino.launch.py [hunk 4]
```diff
@@ -41,6 +41,9 @@
 
 def generate_launch_description():
 
+    # Obtains andino_description's share directory path.
+    pkg_andino_description = get_package_share_directory('andino_description')
```
Comment (jballoffet) [inline andino_description/launch/view_andino.launch.py:45]: Ditto.

File: andino_description/urdf/include/andino_control.urdf.xacro [hunk 5]
```diff
@@ -1,34 +1,43 @@
...
+
+    <!-- TODO(francocipollone): This xacro should be part of andino_control package instead.-->
+    <ros2_control name="RealRobot" type="system">
+      <hardware>
+        <plugin>andino_base/DiffDriveAndino</plugin>
+        <param name="left_wheel_name">${hardware_props['left_wheel_name']}</param>
+        <param name="right_wheel_name">${hardware_props['right_wheel_name']}</param>
+        <param name="serial_device">${hardware_props['serial_device']}</param>
+        <param name="baud_rate">${hardware_props['baud_rate']}</param>
+        <param name="timeout">1${hardware_props['timeout']}</param>
```
Comment (jballoffet) [inline andino_description/urdf/include/andino_control.urdf.xacro:20]: Is the `1` character a typo?

**Review by jballoffet — CHANGES_REQUESTED** [review]
Overall changes LGTM, I left some comments, PTAL. Thanks!

File: andino_description/urdf/include/andino_control.urdf.xacro (hunk 5 above)
Comment (JesusSilvaUtrera) [reply andino_description/urdf/include/andino_control.urdf.xacro:20]: Indeed, I didn't notice it as no error was being raised, it's weird jaja

**Comment by JesusSilvaUtrera** [comment]
> Overall changes LGTM, I left some comments, PTAL. Thanks!

One little question: Franco left a comment about maybe moving the `andino_control.urdf.xacro` to `andino_control` package, but I think it is better to leave it here, as that would create a dependency between `andino_description` and `andino_control`. wdyt?

**Comment by jballoffet** [comment]
> > Overall changes LGTM, I left some comments, PTAL. Thanks!
> 
> One little question: Franco left a comment about maybe moving the `andino_control.urdf.xacro` to `andino_control` package, but I think it is better to leave it here, as that would create a dependency between `andino_description` and `andino_control`. wdyt?

Good question. Let's keep it as is on this PR and if needed, send another PR for moving the xacro file to the `andino_control` package.

**Review by jballoffet — APPROVED** [review]
LGTM, thanks @JesusSilvaUtrera!

**Pull request merged**